backend/workstation/ws_controllers.js

```diff
@@ -0,0 +1,224 @@
+const axios = require("axios");
+const prisma = require("../prisma/prisma_client");
+require("dotenv").config();
+
+//Create New File
+exports.createNewFile = async (req, res) => {
+  const { groupId } = req.params;
+  const { newFileName } = req.body;
+  const userId = req.user.id;
+
+  if (!newFileName) {
+    return res.status(400).json({
+      error: "File name is required",
+    });
+  }
+
+  try {
+    const group = await prisma.codeGroup.findUnique({
+      where: {
+        id: parseInt(groupId),
+      },
+      include: {
+        members: true,
+      },
+    });
+    if (!group) {
+      return res.status(404).json({
+        error: "Group not found",
+      });
+    }
+    const isMember = group.members.some((member) => member.userId === userId);
+    if (!isMember && group.creatorId !== userId) {
+      return res.status(403).json({
+        error: "You do not have access to this group",
+      });
+    }
+    const newFile = await prisma.file.create({
+      data: {
+        fileName: newFileName,
+        user: {
```

Comment:
Not sure if it's too late to change this in the schema, but if this represent the file creator I would recommend naming it something like creator or owner. `user` is vague naming

Reply:
Okay, I will make a new PR for the new prisma schema update and this fix will be included